commit e33536ae131e31adf77d3258c0cdd3da10868534
Author: Kristaps Dz <kristaps@bsd.lv>
Date:   Thu Nov 4 12:27:21 2021 +0000

    Generate a shared library as well.
    
    Earlier, the library interface was just moving too quickly.  Now, I
    don't think I'll be changing it much---if at all, unless to add new
    front-ends.
    
    References https://github.com/kristapsdz/lowdown/issues/45

diff --git a/Makefile b/Makefile
index 475f780..f13a364 100644
--- a/Makefile
+++ b/Makefile
@@ -118,7 +118,7 @@ REGRESS_ARGS	+= "--parse-no-autolink"
 REGRESS_ARGS	+= "--parse-no-cmark"
 REGRESS_ARGS	+= "--parse-no-deflists"
 
-all: lowdown lowdown-diff lowdown.pc
+all: lowdown lowdown-diff liblowdown.so lowdown.pc
 
 www: $(HTMLS) $(PDFS) $(THUMBS) lowdown.tar.gz lowdown.tar.gz.sha512
 
@@ -139,6 +139,10 @@ lowdown-diff: lowdown
 liblowdown.a: $(OBJS) $(COMPAT_OBJS)
 	$(AR) rs $@ $(OBJS) $(COMPAT_OBJS)
 
+
+liblowdown.so: $(OBJS) $(COMPAT_OBJS)
+	$(CC) -shared -o $@ $(OBJS) $(COMPAT_OBJS) $(LDFLAGS)
+
 install: all
 	mkdir -p $(DESTDIR)$(BINDIR)
 	mkdir -p $(DESTDIR)$(LIBDIR)/pkgconfig
@@ -150,6 +154,7 @@ install: all
 	$(INSTALL_PROGRAM) lowdown $(DESTDIR)$(BINDIR)
 	$(INSTALL_PROGRAM) lowdown-diff $(DESTDIR)$(BINDIR)
 	$(INSTALL_LIB) liblowdown.a $(DESTDIR)$(LIBDIR)
+	$(INSTALL_LIB) liblowdown.so $(DESTDIR)$(LIBDIR)
 	$(INSTALL_DATA) lowdown.h $(DESTDIR)$(INCLUDEDIR)
 	for f in $(MANS) ; do \
 		name=`basename $$f .html` ; \
@@ -267,7 +272,7 @@ main.o: lowdown.h
 
 clean:
 	rm -f $(OBJS) $(COMPAT_OBJS) main.o
-	rm -f lowdown lowdown-diff liblowdown.a lowdown.pc
+	rm -f lowdown lowdown-diff liblowdown.a liblowdown.so lowdown.pc
 	rm -f index.xml diff.xml diff.diff.xml README.xml lowdown.tar.gz.sha512 lowdown.tar.gz
 	rm -f $(PDFS) $(HTMLS) $(THUMBS)
 	rm -f index.latex.aux index.latex.latex index.latex.log index.latex.out
